src/components/VirtualList/VirtualItem.tsx

```diff
@@ -0,0 +1,202 @@
+      const istioType = type + 's';
+      const subtype =
+        type === 'adapter' ? this.props.item['adapter']!.adapters : this.props.item['template']!.templates;
+      to = to + '/' + istioType + '/' + subtype + '/' + name;
+    } else {
+      to = to + '/' + dicIstioType[this.props.config.istioTypes[type].name] + '/' + name;
+    }
+    return to;
+  };
+
+  renderDefinition = () => {
+    return (
+      <div className={'list-group-item-heading'}>
+        <div className="ObjectDefinition-Heading">
+          <div className="ObjectDefinition-Title">
+            {this.props.item['name']}
+            <small>{this.props.item.namespace}</small>
+          </div>
+        </div>
+      </div>
+    );
+  };
+
+  renderHealth = () => {
+    return (
+      this.state.health && (
+        <Col xs={12} sm={12} md={4} lg={4}>
+          <strong>Health: </strong>
+          <HealthIndicator id={this.props.item.name} health={this.state.health} mode={DisplayMode.SMALL} />
+        </Col>
+      )
+    );
+  };
+
+  renderSidecar = () => {
+    const istioSidecar = !this.props.item['istioSidecar'];
+    return (
+      <Col xs={12} sm={12} md={4} lg={4}>
+        {istioSidecar && <MissingSidecar />}
+      </Col>
+    );
+  };
+
+  renderLabelValidation = () => {
+    const object = this.props.item;
+    const appLabel = object['appLabel'];
+    const versionLabel = object['versionLabel'];
+    return (
+      <Col xs={12} sm={12} md={4} lg={4}>
+        {appLabel || versionLabel ? (
+          <span>
+            <strong>Label Validation :</strong>
+            {appLabel && <Badge>app</Badge>}
+            {versionLabel && <Badge>version</Badge>}
+          </span>
+        ) : (
+          <span />
+        )}
+      </Col>
+    );
+  };
+
+  renderIstioType = () => {
+    const type = this.props.item['type'];
+    const object = this.props.config.istioTypes[this.props.item['type']];
+    return (
+      <Col xs={12} sm={12} md={4} lg={4}>
+        {type === 'adapter' || type === 'template' ? `${object.name}: ${this.props.item[type][type]}` : object.name}
+      </Col>
+    );
+  };
+
+  renderConfiguration = () => {
+    const object = this.props.item;
+    const validation = object['validation'];
+    return (
+      validation && (
+        <Col xs={12} sm={12} md={4} lg={4}>
+          <strong>Config: </strong>{' '}
+          <ConfigIndicator id={object.name + '-config-validation'} validations={[validation]} size="medium" />
+        </Col>
+      )
+    );
+  };
+
+  renderDetails = () => {
+    return this.props.config.information.map(object => this['render' + object.name]());
+  };
+
+  getIcon = () => {
+    if (this.props.config.name !== 'istio') {
+      return this.props.config.icon;
+    } else {
+      return this.props.config.istioTypes[this.props.item['type']].icon;
+    }
+  };
+
+  render() {
+    const { item, index, config, style, className } = this.props;
+    const ns = this.props.item.namespace;
+    return (
+      <Link
```

Comment:
this link triggers a warning in the console : `Warning: validateDOMNesting(...): <a> cannot appear as a child of <tbody>`
... `in Link (at VirtualItem.tsx:177)`

Reply:
I know I need to check how to deal with this...virtualtable is not working correctly with listview item.